File: 20220720/모법답안/20220720_08.js/transform.py
def solution(ini, trans):
    '''
    :param ini: string
    :param trans: string
    :return: string
    '''
    if ini.count('y') != trans.count('y'):
        return "NO"

    j = 0
    for i in range(len(ini)):
        if ini[i] == 'y':
            continue
        while trans[j] == 'y':
            j += 1
        if ini[i] != trans[j] or (ini[i] == 'x' and i > j) or (ini[i] == 'z' and i < j):
            return "NO"
        j += 1
    else:
        return "YES"

File: 20220720/모법답안/20220720_08.js/transform.js
/**
 * @param {string} ini
 * @param {string} trans
 * @return {string}
 */
 function solution(ini, trans) {
    let i = 0;
    let j = 0;

    while (i < ini.length || j < trans.length) {
        while (ini[i] == "y") i++;
        while (trans[j] == "y") j++;

        if ((i == ini.length || j == ini.length) && i != j) return "NO";
        if (ini[i] !== trans[j]) return "NO";
        if (ini[i] == "x" && i > j) return "NO";
        if (ini[i] == "z" && i < j) return "NO";

        i++;
        j++;
    }

    return "YES";
}
